src/components/bmfont-text.js

```diff
@@ -0,0 +1,261 @@
+};
+
+module.exports.Component = registerComponent('bmfont-text', {
+  schema: {
+    // scale is now determined by width and wrappixels/wrapcount... scale: {default: 0.003},
+    font: {default: ''},
+    tabSize: {default: 4},
+    anchor: {default: 'center', oneOf: anchors}, // center default to match primitives like plane; if 'align', null or undefined, same as align
+    baseline: {default: 'center', oneOf: baselines},
+    text: {type: 'string'},
+    width: {type: 'number'}, // use AFRAME units i.e. meters, not arbitrary numbers... // default to geometry width, or if not present then DEFAULT_WIDTH
+    height: {type: 'number'}, // use AFRAME units i.e. meters, not arbitrary numbers... // no default, will be populated at layout
+    align: {type: 'string', default: 'left', oneOf: alignments},
+    letterSpacing: {type: 'number', default: 0},
+    lineHeight: {type: 'number'},  // default to font's lineHeight value
+    fnt: {type: 'string', default: 'default'},
+    fntImage: {type: 'string'}, // default to fnt but with .fnt replaced by .png
+    mode: {default: 'normal', oneOf: ['normal', 'pre', 'nowrap']},
+    color: {type: 'color', default: '#000'},
+    opacity: {type: 'number', default: '1.0'},
+    type: {default: 'SDF', oneOf: ['SDF', 'basic', 'MSDF']},
+    side: {default: 'front', oneOf: ['front', 'back', 'double']},
+    transparent: {default: true},
+    alphaTest: {default: 0.5},
+    wrappixels: {type: 'number'}, // if specified, units are bmfont pixels (e.g. DejaVu default is size 32)
+    wrapcount: {type: 'number', default: 40} // units are 0.6035 * font size e.g. about one default font character (monospace DejaVu size 32)
+  },
+
+  init: function () {
+    this.texture = new THREE.Texture();
+    this.texture.anisotropy = 16; // ??
+
+    this.geometry = createTextGeometry();
+
+    this.updateMaterial();
+    this.mesh = new THREE.Mesh(this.geometry, this.material);
+    this.el.setObject3D('bmfont-text', this.mesh);
+  },
+
+  update: function (oldData) {
+    var data = this.coerceData(this.data);
+
+    // decide whether to update font, or just text data
+    if (!oldData || oldData.fnt !== data.fnt) {
+      // new font, will also subsequently change data & layout
+      this.updateFont();
+    } else if (this.currentFont) {
+      // new data like change of text string
+      var font = this.currentFont;
+      var textRenderWidth = data.wrappixels || (data.wrapcount * 0.6035 * font.info.size);
+      var options = assign({}, data, { font: font, width: textRenderWidth, lineHeight: data.lineHeight || font.common.lineHeight });
+      this.geometry.update(options);
+      this.updateLayout(data);
+    }
+    // ??
+    this.updateMaterial(oldData.type);
+  },
+
+  remove: function () {
+    this.geometry.dispose();
+    this.geometry = null;
+    this.el.removeObject3D('bmfont-text');
+  },
+
+  coerceData: function (data) {
+    // We have to coerce some data to numbers/booleans
```

Comment:
looking at this slightly more, I think the data needs to be coerced because it goes right into the text creation in the options merge.